File: app/src/main/java/com/prox/voicechanger/database/FileVoiceDAO.java
package com.prox.voicechanger.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.prox.voicechanger.model.FileVoice;

import java.util.List;

@Dao
public interface FileVoiceDAO {
    @Insert
    void insert(FileVoice fileVoice);

    @Update
    void update(FileVoice fileVoice);

    @Delete
    void delete(FileVoice fileVoice);

    @Query("SELECT * FROM filevoice WHERE path LIKE '%mp3'")
    List<FileVoice> getAllVoice();

    @Query("SELECT * FROM filevoice WHERE path LIKE '%mp4'")
    List<FileVoice> getAllVideo();

    @Query("SELECT * FROM filevoice WHERE path= :path")
    FileVoice check(String path);
}

File: app/src/main/java/com/prox/voicechanger/database/FileVoiceDAO.kt
package com.prox.voicechanger.database

import androidx.annotation.Nullable
import androidx.room.*
import com.prox.voicechanger.model.FileVoice

@Dao
interface FileVoiceDAO {
    @Insert
    fun insert(fileVoice: FileVoice?)

    @Update
    fun update(fileVoice: FileVoice?)

    @Delete
    fun delete(fileVoice: FileVoice?)

    @get:Query("SELECT * FROM filevoice WHERE path LIKE '%mp3'")
    val allVoice: List<FileVoice?>?

    @get:Query("SELECT * FROM filevoice WHERE path LIKE '%mp4'")
    val allVideo: List<FileVoice?>?

    @Query("SELECT * FROM filevoice WHERE path= :path")
    fun check(path: String?): FileVoice?
}